morango/models/fields/uuids.py

```diff
@@ -2,67 +2,35 @@
 import uuid
 
 from django.db import models
+
 from morango.utils import _assert
 
 
 def sha2_uuid(*args):
     return hashlib.sha256("::".join(args).encode("utf-8")).hexdigest()[:32]
 
 
-class UUIDField(models.CharField):
+class UUIDField(models.UUIDField):
     """
     Adaptation of Django's UUIDField, but with 32-char hex representation as Python representation rather than a UUID instance.
     """
 
-    def __init__(self, *args, **kwargs):
-        kwargs["max_length"] = 32
-        super(UUIDField, self).__init__(*args, **kwargs)
-
-    def prepare_value(self, value):
-        if isinstance(value, uuid.UUID):
-            return value.hex
-        return value
-
-    def deconstruct(self):
-        name, path, args, kwargs = super(UUIDField, self).deconstruct()
-        del kwargs["max_length"]
-        return name, path, args, kwargs
-
-    def get_internal_type(self):
-        return "UUIDField"
-
     def get_db_prep_value(self, value, connection, prepared=False):
         if value is None:
             return None
         if not isinstance(value, uuid.UUID):
-            try:
-                value = uuid.UUID(value)
-            except AttributeError:
-                raise TypeError(self.error_messages["invalid"] % {"value": value})
+            value = super(UUIDField, self).to_python(value)
+
+        if connection.features.has_native_uuid_field:
+            return value
         return value.hex
```

Comment:
I'm now not seeing a difference between this and the parent class implementation, other than the use of `super`, which doesn't seem necessary, since `to_python` isn't defined in this class, so it'll use the super's anyway. So I'm guessing this can be removed?

Reply:
It is, see below.